src/Control/Email/Email.php

```diff
@@ -769,237 +399,150 @@ public function getHTMLTemplate()
         }
         $this->HTMLTemplate = $template;
-
         return $this;
     }
 
     /**
      * Get the template to render the plain part with
-     *
-     * @return string
      */
-    public function getPlainTemplate()
+    public function getPlainTemplate(): string
     {
         return $this->plainTemplate;
     }
 
     /**
      * Set the template to render the plain part with
-     *
-     * @param string $template
-     * @return $this
      */
-    public function setPlainTemplate($template)
+    public function setPlainTemplate(string $template): static
     {
         if (substr($template ?? '', -3) == '.ss') {
             $template = substr($template ?? '', 0, -3);
         }
         $this->plainTemplate = $template;
-
-        return $this;
-    }
-
-    /**
-     * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
-     *
-     * @param array $recipients
-     * @return $this
-     */
-    public function setFailedRecipients($recipients)
-    {
-        Deprecation::notice('4.12.0', 'Will be removed without equivalent functionality to replace it');
-        $this->failedRecipients = $recipients;
-
         return $this;
     }
 
-    /**
-     * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
-     *
-     * @return array
-     */
-    public function getFailedRecipients()
-    {
-        Deprecation::notice('4.12.0', 'Will be removed without equivalent functionality to replace it');
-        return $this->failedRecipients;
-    }
-
-    /**
-     * @deprecated 4.12.0 Will be replaced with getData()
-     *
-     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
-     *
-     * @return bool
-     */
-    public function IsEmail()
-    {
-        Deprecation::notice('4.12.0', 'Will be replaced with getData()');
-        return true;
-    }
-
     /**
      * Send the message to the recipients
-     *
-     * @return bool true if successful or array of failed recipients
      */
-    public function send()
+    public function send(): void
     {
-        if (!$this->getBody()) {
-            $this->render();
-        }
-        if (!$this->hasPlainPart()) {
-            $this->generatePlainPartFromBody();
-        }
-        return Injector::inst()->get(Mailer::class)->send($this);
+        $this->updateHtmlAndTextWithRenderedTemplates();
+        Injector::inst()->get(MailerInterface::class)->send($this);
     }
 
     /**
-     * @return array|bool
+     * Send the message to the recipients as plain-only
      */
-    public function sendPlain()
+    public function sendPlain(): void
     {
-        if (!$this->hasPlainPart()) {
-            $this->render(true);
-        }
-        return Injector::inst()->get(Mailer::class)->send($this);
+        $html = $this->getHtmlBody();
+        $this->html(null);
+        $this->updateHtmlAndTextWithRenderedTemplates(true, $html);
```

Comment:
```suggestion
        $this->updateHtmlAndTextWithRenderedTemplates(true);
        $this->html(null);
```
It looks like the html is only being set to null to avoid sending the html version if one exists - in which case this change removes the need for it altogether.

If the fallback is for some additional purpose, it should be added as a comment in the phpdoc for `updateHtmlAndTextWithRenderedTemplates()`.

Reply:
We need the HTML fallback in case we are sending a plain email only, though text() has not be called.  In that scenario the plain email is created with `DBField::create_field('HTMLFragment', html_entity_decode($htmlRender))->Plain()` which is the html render with html tags stripped

Reply:
In that scenario, won't my above suggested change mean that `$htmlBody = $this->getHtmlBody()` in `updateHtmlAndTextWithRenderedTemplates()` gets the body anyway? So in that case we don't need a _fallback_ because we're already using the _actual html body_?